.buildkite/scripts/test/set-deployer-config.sh

```diff
@@ -51,8 +51,10 @@ write_deployer_config() {
     w '  kubernetesVersion: "'"${DEPLOYER_K8S_VERSION}"'"'
     fi
 
-    case "$E2E_PROVIDER" in gke|ocp)
-    w "  $E2E_PROVIDER:"
+    case "$E2E_PROVIDER" in gke|ocp|gke-autopilot)
+    # stripping "-autopilot" if it exists from the e2e provider
+    # as the providers do not include the "autopilot" suffix.
+    w "  ${E2E_PROVIDER%"-autopilot"}:"
```

Comment:
I think it would be good to do this in a more generic way adopting the convention that the provider is the first element separated by `-` in an e2e provider:
```suggestion
    # extract provider to configure from the e2e provider (first elem separated by '-')
    read -r -d '-' provider <<< "$E2E_PROVIDER"
    case "$provider" in gke|ocp)
    w "  $provider:"
```

Reply:
Last build failed because `read -r -d '-' provider <<< "$E2E_PROVIDER"` returns `1` when `E2E_PROVIDER` doesn't contain a `-`, which is the case for PR where it is set to `kind`.

Proposal for a new syntax:
```diff
diff --git a/.buildkite/scripts/test/set-deployer-config.sh b/.buildkite/scripts/test/set-deployer-config.sh
index 6fc6a44b2..0d61fb453 100755
--- a/.buildkite/scripts/test/set-deployer-config.sh
+++ b/.buildkite/scripts/test/set-deployer-config.sh
@@ -18,7 +18,7 @@
 #   DEPLOYER_KIND_NODE_IMAGE
 #   DEPLOYER_KIND_IP_FAMILY

-set -eux
+set -eu

 WD="$(cd "$(dirname "$0")"; pwd)"
 ROOT="$WD/../../.."
@@ -51,10 +51,9 @@ write_deployer_config() {
     w '  kubernetesVersion: "'"${DEPLOYER_K8S_VERSION}"'"'
     fi

+    case "$E2E_PROVIDER" in gke*|ocp*)
     # extract provider to configure from the e2e provider (first elem separated by '-')
-    read -r -d '-' provider <<< "$E2E_PROVIDER"
-    case "$provider" in gke|ocp)
-    w "  $provider:"
+    w "  ${E2E_PROVIDER%%-*}:"
     w "    gCloudProject: elastic-cloud-dev"
     ;; esac
```